config/servicemappings/vertexai.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: core.cnrm.cloud.google.com/v1alpha1
+kind: ServiceMapping
+metadata:
+  name: vertexai.cnrm.cloud.google.com
+  namespace: cnrm-system
+spec:
+  name: VertexAI
+  version: v1beta1
+  serviceHostName: "aiplatform.googleapis.com"
+  resources:
+    - name: google_vertex_ai_dataset
+      kind: VertexAIDataset
+      # This resource does not support import.
+      skipImport: true
+      idTemplate: "{{name}}"
+      idTemplateCanBeUsedToMatchResourceName: false
+      resourceAvailableInAssetInventory: true
+      v1alpha1ToV1beta1: true
+      storageVersion: v1alpha1
+      metadataMapping:
+        labels: labels
+      serverGeneratedIDField: name
+      resourceID:
+        targetField: name
```

Comment:
Just to confirm, the name field doesn't have a format (e.g. "projects/[PROJECT_ID]")?

Reply:
This is what I got when playing with TF directly. It looks to me that the TF field `name` already has all the information?

```
$ terraform state show google_vertex_ai_dataset.dataset
# google_vertex_ai_dataset.dataset:
resource "google_vertex_ai_dataset" "dataset" {
    create_time         = "2024-02-16T23:56:54.194318Z"
    display_name        = "terraform"
    id                  = "projects/674741657116/locations/us-central1/datasets/5201786212473372672"
    labels              = {
        "env" = "test"
    }
    metadata_schema_uri = "gs://google-cloud-aiplatform/schema/dataset/metadata/image_1.0.0.yaml"
    name                = "projects/674741657116/locations/us-central1/datasets/5201786212473372672"
    project             = "cnrm-jingyih"
    region              = "us-central1"
    update_time         = "2024-02-16T23:57:43.624186Z"
}
```

Reply:
In that case a valueTemplate will be needed for resourceID configuration.
Example: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/servicemappings/monitoring.yaml#L30
The format of the target field of the example configuration can be found in the TF field description: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/monitoring_alert_policy#attributes-reference

Reply:
I see. I realized that I had misunderstood the mapping direction; it was actually the opposite of what I initially thought... 
But for this particular resource, both `resourceID` and `name` has the same format and value. Why is a valueTemplate still needed?

Reply:
Discussed offline with Joyce. The valueTemplate is indeed necessary. Without valueTemplate, the `spec.resourceID` will look like `projects/674741657116/locations/us-central1/datasets/7410406005512601600`, which is not what we want. The correct resourceID value should only contain the last part of the string. e.g. `7410406005512601600`.